src/plugins/mer/mersettings.cpp

```diff
@@ -211,42 +213,147 @@ FileName MerSettings::globalDeviceModelsFileName()
                                 + QLatin1String(MER_DEVICE_MODELS_FILENAME));
 }
 
+FileName MerSettings::deviceModelsFileName()
+{
+    const QFileInfo settingsLocation(PluginManager::settings()->fileName());
+    return FileName::fromString(settingsLocation.absolutePath() + QLatin1String(MER_DEVICE_MODELS_FILENAME));
+}
+
 QMap<QString, MerEmulatorDeviceModel> MerSettings::deviceModels()
 {
     Q_ASSERT(s_instance);
 
     return s_instance->m_deviceModels;
 }
 
-void MerSettings::read()
```

Comment:
Please try to keep read() defined here (and define the new functions first) to minimize the diff.

Reply:
Reworked. Diff is minimal.